File: ios/CheckSimulator.m
//
//  CheckSimulator.m
//  NativeModulesIntro
//
//  Created by Dipti Agrawal on 10/08/21.
//

#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"


@interface RCT_EXTERN_MODULE(CheckSimulator, NSObject)
RCT_EXTERN_METHOD(isSimulator:(RCTResponseSenderBlock *)successCallback)

@end

File: ios/CheckSimulator.swift
//
//  CheckSimulator.swift
//  NativeModulesIntro
//
//  Created by Dipti Agrawal on 10/08/21.
//

import Foundation

@objc(CheckSimulator)
class checkSimulator: NSObject {
  
  @objc
  func isSimulator(_ successCallback: RCTResponseSenderBlock){
    if(TARGET_OS_SIMULATOR != 0){
      successCallback([NSNull(),true])
    }
    else
    {
      successCallback([NSNull(),false])
    }
       
    }

}
